bakta/features/cds.py

```diff
@@ -166,75 +167,69 @@ def create_cdss(genes, contig):
-                domain_stop = int(cols[16])
-                domain_cov = (domain_stop - domain_start + 1) / domain_length
-                aa_start = int(cols[19])
-                aa_stop = int(cols[20])
-                aa_cov = (aa_stop - aa_start + 1) / len(cds['aa'])
-
-                pfam = OrderedDict()
-                pfam['id'] = cols[4]
-                pfam['name'] = cols[3]
-                pfam['length'] = domain_length
-                pfam['aa_cov'] = aa_cov
-                pfam['hmm_cov'] = domain_cov
-                pfam['evalue'] = float(cols[6])
-                pfam['score'] = float(cols[7])
-                pfam['start'] = aa_start
-                pfam['stop'] = aa_stop
-
-                if('pfams' not in cds):
-                    cds['pfams'] = []
-                cds['pfams'].append(pfam)
-                if('db_xrefs' not in cds):
-                    cds['db_xrefs'] = []
-                cds['db_xrefs'].append(f"PFAM:{pfam['id']}")
-                pfam_hits.append(cds)
-                cds_pfams_hits[aa_identifier] = cds
-                log.info(
-                    'pfam detected: contig=%s, start=%i, stop=%i, strand=%s, pfam-id=%s, length=%i, aa-start=%i, aa-stop=%i, aa-cov=%1.1f, hmm-cov=%1.1f, evalue=%1.1e, bitscore=%1.1f, name=%s',
-                    cds['contig'], cds['start'], cds['stop'], cds['strand'], pfam['id'], pfam['length'], pfam['start'], pfam['stop'], pfam['aa_cov'], pfam['hmm_cov'], pfam['evalue'], pfam['score'], pfam['name']
-                )
+    for hit in hits:
+        cds = orf_by_aa_digest[hit['query']]
+
+        domain_length = hit['domain_length']
+        domain_start = hit['domain_start']
+        domain_stop = hit['domain_stop']
+        domain_cov = (domain_stop - domain_start + 1) / domain_length
+        aa_start = hit['aa_start']
+        aa_stop = hit['aa_stop']
+        aa_cov = (aa_stop - aa_start + 1) / len(cds['aa'])
+
+        pfam = OrderedDict()
+        pfam['name'] = hit['subject_name']
+        pfam['id'] = hit['subject_id']
+        pfam['length'] = domain_length
+        pfam['aa_cov'] = aa_cov
+        pfam['hmm_cov'] = domain_cov
+        pfam['evalue'] = hit['evalue']
+        pfam['score'] = hit['bitscore']
+        pfam['start'] = aa_start
+        pfam['stop'] = aa_stop
+
+        cds.setdefault('pfams', [])
+        cds['pfams'].append(pfam)
+        cds.setdefault('db_xrefs', [])
+        cds['db_xrefs'].append(f"PFAM:{pfam['id']}")
+        pfam_hits.append(cds)
+        cds_pfams_hits.append(cds)
+        log.info(
+            'pfam detected: contig=%s, start=%i, stop=%i, strand=%s, pfam-id=%s, length=%i, aa-start=%i, aa-stop=%i, aa-cov=%1.1f, hmm-cov=%1.1f, evalue=%1.1e, bitscore=%1.1f, name=%s',
+            cds['contig'], cds['start'], cds['stop'], cds['strand'], pfam['id'], pfam['length'], pfam['start'], pfam['stop'], pfam['aa_cov'], pfam['hmm_cov'], pfam['evalue'], pfam['score'], pfam['name']
+        )
```

Comment:
Please integrate 2nd loop over `hits` (ll 199-230) with 1st (ll 178-194). Hits can be read, filtered and stored in `pfam_hits` immediately, so `hits` is not required.